core/spec/services/spree/stock_locations/stock_items/create_spec.rb

```diff
@@ -0,0 +1,99 @@
+require 'spec_helper'
+
+module Spree
+  describe StockLocations::StockItems::Create do
+    subject { described_class }
+
+    let!(:stock_location) { create(:stock_location_with_items) }
+    let!(:unrelated_variant) { create(:variant) }
+    let(:result) { subject.call(stock_location: stock_location) }
+    let(:stock_location_class) { stock_location.class }
+
+    describe '#call' do
+      context 'Rails < 6', if: Rails::VERSION::MAJOR < 6 do
+        context 'with variants to propagate' do
+          before { stock_location.stock_items.delete_all }
+
+          it 'propagates the variants' do
+            expect { result }.to change { stock_location.stock_items.count }.from(0).to(4)
+          end
+        end
+
+        context 'without variants to propagate' do
+          before { Variant.destroy_all }
+
+          it 'does not propagate stock location variants' do
+            expect(stock_location).not_to receive(:propagate_variant)
+            result
+          end
+        end
+      end
+
+      context 'Rails >= 6', if: Rails::VERSION::MAJOR >= 6 do
+        before do
+          # Delete all stock_location.stock_items to start counting from 0.
+          stock_location.stock_items.unscope(:where).delete_all
+        end
+
+        let(:time_current) { Time.local(1990) }
+
+        context 'with prepared stock items' do
+          context 'with duplicate stock items by stock_location_id and variant' do
+            it 'removes duplicate stock_items before inserting to avoid duplicate ones' do
```

Comment:
@sebastian-palma this description is kinda cryptic - does it remove the previous records or just skip the duplicates? :) auto-removing stuff makes my skin crawl 😂

Reply:
> does it remove the previous records or just skip the duplicates

It kind of skips them. It doesn't touch the db before that, that's certainly a confusing description.

I wrote it looking at [these](https://github.com/spree/spree/blob/a46b93ea1088db99b6f3b921e6f92af437971294/core/app/services/spree/stock_locations/stock_items/create.rb#L9:L17) lines. As `prepared_stock_items` doesn't include the `deleted_at` values, the index we just created isn't used for the insertion, so we have to remove the duplicates by hand and then insert them

Reply:
@sebastian-palma  Let's change that description - it's misleading ATM